Blob Update

alisa-ulan opened the pull request:
A blob who is in space/solars/off the z-level when the blobs burst will be warned and given 30 seconds to get back on the station.  If he doesn't, he'll gib, and a new crewmember will be infected.  Feedback on the specifics of the admin logging/messaging is welcome.

alisa-ulan committed:
Blob Update

Blobs' bursting on solars makes a log and an admin message, and will end
the round if the blob bursting was the last remaining blob.  Feedback on
the specifics of the admin logging/messaging is welcome.

Iamgoofball commented:
make this either A. respawn them at a blobspawn position

or B. make a new blob at a blobspawn position

Deantwo commented:
Doesn't #7418 fix this?

Dorsisdwarf commented:
The whole point of this PR is that #7418 doesn't solve this in the correct manner, and paprika doesn't want to work on reblob himself.

Deantwo commented:
Yeah I was wondering why the whole round should end.

paprka commented:
Because wizard ends when the wizard nukes himself anyway

The round needs to end because you can't call the fucking shuttle during blob rounds and my solution was the easiest

alisa-ulan committed:
Makes blob more sensible

Gives blobs who burst in space a 30 second warning before they're
killed, and informs the admins that it's about to happen.  Also changed
the round ending logic to use the check_finished proc, rather than
building it in to the blobize proc.

AlexanderUlanH commented:
Replacement spawning is coming, but there's no good way to do it with blobcode at the moment, so it'll take some refactoring.

alisa-ulan committed:
Reblob

A blob bursting in space will cause a random valid crewmember to be
infected.  They'll have a short time to get to a good location before
they burst.  This means a lone blob bursting in space will not end the
round unless there are no blob-valid crewmembers to choose from.

AlexanderUlanH commented:
The procs are flexible enough that it should be pretty easy to make a one-click-admin for blob now, if someone wants to do that.

alisa-ulan committed:
Fixed Typo

Very important

Iamgoofball commented:
okay, now it's :+1:

Jordie0608 commented:
Changelog.

alisa-ulan committed:
Changelog

Logged changes
